test/functional/viml/completion_spec.lua

```diff
@@ -1072,4 +1072,51 @@ describe('completion', function()
       set complete&vim completeopt&vim
     ]])
   end)
+
+  it('MenuPopupChanged autocommand', function()
+    curbufmeths.set_lines(0, 1, false, { 'foo', 'bar', 'foobar', ''})
+    source([[
+      set complete=. completeopt=noinsert,noselect,menuone
+      function! OnPumChange()
+        let g:event = copy(v:event)
+        let g:item = get(v:event, 'completed_item', {})
+        let g:word = get(g:item, 'word', v:null)
+      endfunction
+      autocmd! MenuPopupChanged * :call OnPumChange()
+      call cursor(4, 1)
+    ]])
+
+    feed('S')
+    wait()
```

Comment:
This is a typical place where a screen test should be used instead of `wait()`.  Screen is already initialized in this describe block, so just replace every wait with `screen:snapshot_util()`.  Though some could probably just be eliminated, I think this can be `feed('Sf<c-N>`)` and then a screen test.